packages/k8s/src/k8s/index.ts

```diff
@@ -369,6 +383,22 @@ export async function waitForPodPhases(
   }
 }
 
+export function getPrepareJobTimeoutSeconds(): number {
+  const envTimeoutSeconds =
+    process.env['ACTIONS_RUNNER_PREPARE_JOB_TIMEOUT_SECONDS']
+
+  if (!envTimeoutSeconds) {
+    return DEFAULT_WAIT_FOR_POD_TIME_SECONDS
+  }
+
+  const timeoutSeconds = parseInt(envTimeoutSeconds, 10)
+  if (!timeoutSeconds) {
+    return DEFAULT_WAIT_FOR_POD_TIME_SECONDS
```

Comment:
The only thing I would like to add here is maybe check `timeoutSeconds > 0` and add a warning. Otherwise, looks great